basic/atm/atm_exercise.py

```diff
@@ -0,0 +1,208 @@
+    customers_data[customer_id][BALANCE_KEY] = new_balance
+    print(f"withdrawal was successful {customer_id} now has {new_balance}$.")
+
+
+def change_password(customers_data, customer_id):
+    """Change the given customer's password.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+    """
+    new_pass = input("Enter a new password.\n")
+    customers_data[customer_id][PASS_KEY] = new_pass
+    print(f"Password was changed successfully")
+
+
+def get_data_from_file(filename):
+    """Get bank details from file, arrange and return them in dictionary.
+
+    Args:
+        filename (str): The path to the atm details file.
+
+    Returns:
+         dict: Dictionary of all customer's data from the file.
+    """
+    with open(filename, newline='') as csv_atm_file:
+        customers = {}
+        customers_info = csv.reader(csv_atm_file, delimiter=' ', quotechar='|')
+        for line in customers_info:
+            customer_id, the_pass, balance = line
+            customer = {PASS_KEY: the_pass, BALANCE_KEY: float(balance)}
+            customers[customer_id] = customer
+
+    return customers
+
+
+def execute_choice(customers_data, customer_id, choice):
+    """Execute a function according to the choice given.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+        choice (str): The user's desired function to execute.
+    """
+    if choice == "1":
+        check_balance(customers_data, customer_id)
+
+    elif choice == "2":
+        cash_withdrawal(customers_data, customer_id)
+
+    elif choice == "3":
+        cash_deposit(customers_data, customer_id)
+
+    else:
+        change_password(customers_data, customer_id)
+
+
+def update_file(filename, data):
+    """Update the file with the data in the given dict 'data'.
+
+    Args:
+        filename (str): Path too the atm file.
+        data (dict): Dictionary with data from the atm file.
+    """
+    with open(filename, "w") as atm_file:
+        for key in data:
+            line = f"{key} {data[key][PASS_KEY]} {data[key][BALANCE_KEY]}\n"
+            atm_file.write(line)
+
+
+def operations_for_customer(data, customer_id):
+    """Get desired operation by user and execute it.
+
+    Args:
+        data (dict): Dictionary with data from the atm file.
+        customer_id (str): The specific customer id to work on.
+    """
+    msg = """What operation do you want to do?(-1 to stop)
+        1 to check balance
+        2 to withdrawal
+        3 for cash deposit
+        4 to change password\n"""
+    for operation in iter(partial(input, msg), "-1"):
+        if operation not in "1234":
+            print("you have to enter one of the options, try again.")
+            continue
+
+        execute_choice(data, customer_id, operation)
+
+
+def atm(filename):
```

Comment:
Its easy to steal money from people, you never ask from the user to enter his password to perform an action